dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,770 @@
+
+        return this.InternalDeleteBatchAsync([key], condition, options, cancellationToken);
+    }
+
+    private Task InternalDeleteBatchAsync<TKey>(
+        IEnumerable<TKey> keys,
+        DeleteRecordOptions? options,
+        CancellationToken cancellationToken)
+    {
+        var condition = new SqliteWhereInCondition(
+            this._propertyReader.KeyPropertyStoragePropertyName,
+            keys.Cast<object>().ToList());
+
+        return this.InternalDeleteBatchAsync(keys, condition, options, cancellationToken);
+    }
+
+    private Task InternalDeleteBatchAsync<TKey>(
+        IEnumerable<TKey> keys,
+        SqliteWhereCondition condition,
+        DeleteRecordOptions? options,
+        CancellationToken cancellationToken)
+    {
+        const string OperationName = "Delete";
+
+        var tasks = new List<Task>();
+        var keysList = keys.Select(key => (object)key!).ToList();
+
+        if (this._vectorPropertiesExist)
+        {
+            using var vectorCommand = this._commandBuilder.BuildDeleteCommand(
+                this._vectorTableName,
+                [condition]);
+
+            tasks.Add(this.RunOperationAsync(OperationName, () => vectorCommand.ExecuteNonQueryAsync(cancellationToken)));
+        }
+
+        using var dataCommand = this._commandBuilder.BuildDeleteCommand(
+            this._dataTableName,
+            [condition]);
+
+        tasks.Add(this.RunOperationAsync(OperationName, () => dataCommand.ExecuteNonQueryAsync(cancellationToken)));
+
+        return Task.WhenAll(tasks);
+    }
+
+    private TRecord GetAndMapRecord(
+        string operationName,
+        DbDataReader reader,
+        List<VectorStoreRecordProperty> properties,
+        bool includeVectors)
+    {
+        var storageModel = new Dictionary<string, object?>();
+
+        foreach (var property in properties)
+        {
+            var propertyName = this._propertyReader.GetStoragePropertyName(property.DataModelPropertyName);
+            var propertyType = property.PropertyType;
+            var propertyValue = SqliteVectorStoreRecordPropertyMapping.GetPropertyValue(reader, propertyName, propertyType);
+
+            storageModel.Add(propertyName, propertyValue);
+        }
+
+        return VectorStoreErrorHandler.RunModelConversion(
+            DatabaseName,
+            this.CollectionName,
+            operationName,
+            () => this._mapper.MapFromStorageToDataModel(storageModel, new() { IncludeVectors = includeVectors }));
+    }
+
+    private async Task<T> RunOperationAsync<T>(string operationName, Func<Task<T>> operation)
+    {
+        try
+        {
+            return await operation.Invoke().ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            throw new VectorStoreOperationException("Call to vector store failed.", ex)
+            {
+                VectorStoreType = DatabaseName,
+                CollectionName = this.CollectionName,
```

Comment:
`VectorStoreOperationException` exception doesn't contain a property for a second table. We can include vector table name in a message, but it's not always a case that vector table is used at all. Also, if there is `CollectionName` which is data table name, it's possible to calculate vector table name since it's prefix+data table name. Based on that, I think it's fine to provide data table name only.